mbus-api/src/event_traits.rs

```diff
@@ -0,0 +1,55 @@
+use crate::event::{Component, EventMessage, EventMeta, EventSource, Version};
+use chrono::Utc;
+use once_cell::sync::OnceCell;
+use std::str::FromStr;
+
+/// Once cell static variable to store the component field of the event source.
+static COMPONENT: OnceCell<Component> = OnceCell::new();
+
+/// Initialize the event source component with the service name.
+pub fn initilize_source_component(comp: String) {
+    COMPONENT.get_or_init(|| Component::from_str(&comp).unwrap_or_default());
+}
+
+impl EventMeta {
+    /// New event metadata with default values.
+    pub fn new() -> Self {
+        Self {
+            id: uuid::Uuid::new_v4().to_string(),
+            source: Some(EventSource::new()),
+            timestamp: Some(Utc::now().into()),
+            version: Version::V1.into(),
+        }
+    }
+}
+
+impl EventSource {
+    /// New event source with default values.
+    pub fn new() -> Self {
+        let component = (*COMPONENT.get().unwrap_or(&Component::UnknownComponent)).into();
+        Self {
+            component,
+            node: "".to_string(),
```

Comment:
Should set the node here?

Reply:
I am using the default value of null string here. And setting the node value from the data in event trait. Should I use a param and set the node value here?

Reply:
hmm where is it used, maybe new should be called there with node as parameter?